pkg/ingester/shutdown_marker.go

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ingester
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"time"
+)
+
+type ShutdownMarker struct {
+	Path string
+}
+
+func NewShutdownMarker(path string) *ShutdownMarker {
+	return &ShutdownMarker{path}
+}
+
+func (m *ShutdownMarker) Create() error {
```

Comment:
Planning to add some comments here to explain the tricky-ish order of operations.